c10/cuda/CUDAStream.cpp

```diff
@@ -321,6 +334,43 @@ cudaStream_t CUDAStream::stream() const {
   return ptr->stream;
 }
 
+CUDAAssert* CUDAStream::assert_state() const {
+  auto ptr = CUDAStream_internals(*this);
+  AT_ASSERT(ptr);
+
+  // lazy create assert state
+  CUDAAssert* assert_state = ptr->assert_state.load();
+  if (assert_state == nullptr) {
```

Comment:
Hmm... this feels racy too. What if two threads initialize the assert state at the same time? Then you'll just leak the allocated host memory in this case, right?